File: Javascript/Algorithms/Search/linearSearch.js
function linearSearch(arr, elem) {
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] === elem) {
            return i;
        }
    }
    return -1;
}

// Tests
let numbers = [2, 5, 7, 8, 10, 13, 14];
console.log(linearSearch(numbers, 5)); // -> 1
console.log(linearSearch(numbers, 8)); // -> 3
console.log(linearSearch(numbers, 14)); // -> 6
console.log(linearSearch(numbers, 6)); // -> -1

File: Go/linearSearch.go
package main 

import (
	"fmt"
	"strconv"
)

func main(){
	numbers := []int{5, 7, 8, 10, 45, 66};
	fmt.Println(linearSearch(numbers, 45)); //returns Target 45 is found at index 4
	fmt.Println(linearSearch(numbers, 1));  //returns Target 1 not found at the list
}

func linearSearch(input []int, target int) string{
	for key, value := range input {
		if value == target {
			res := "Target " + strconv.Itoa(target) + " is found at index " + strconv.Itoa(key) 
			return res;
		}
	}
	return "Target " + strconv.Itoa(target) + " not found at the list"
}